lib/usdUfe/ufe/UsdUndoClearDefaultPrimCommand.cpp

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2023 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdUndoClearDefaultPrimCommand.h"
+
+#include <usdUfe/base/tokens.h>
+#include <usdUfe/ufe/Utils.h>
+#include <usdUfe/undo/UsdUndoBlock.h>
+#include <usdUfe/utils/editRouter.h>
+#include <usdUfe/utils/editRouterContext.h>
+
+#include <pxr/usd/usdGeom/imageable.h>
+
+namespace USDUFE_NS_DEF {
+
+UsdUndoClearDefaultPrimCommand::UsdUndoClearDefaultPrimCommand(const UsdPrim& prim)
+    : Ufe::UndoableCommand()
+    , _prim(prim)
+{
+    _stage = _prim.GetStage();
+    if (!_stage) {
+        return;
+    }
+}
+
+UsdUndoClearDefaultPrimCommand::~UsdUndoClearDefaultPrimCommand() { }
+
+void UsdUndoClearDefaultPrimCommand::execute()
+{
+    UsdUndoBlock undoBlock(&_undoableItem);
+
+    // Check if the layer selected is the root layer.
+    if (_stage->GetRootLayer() != _stage->GetEditTarget().GetLayer()) {
```

Comment:
For now this is the only case, so it's fine, but I'd rather have a utility function that checks if metadata is allowed to be edited. So if we need to check it in other case, the code is the same and if we change the restriction rules in the future, there would be only one place to change it.